server/src/main/java/org/elasticsearch/search/aggregations/InternalOrder.java

```diff
@@ -244,34 +223,59 @@ public boolean equals(Object obj) {
             CompoundOrder other = (CompoundOrder) obj;
             return Objects.equals(orderElements, other.orderElements);
         }
+    }
 
-        /**
-         * {@code Comparator} for sorting buckets by multiple criteria.
-         */
-        static class CompoundOrderComparator implements Comparator<Bucket> {
-
-            private List<BucketOrder> compoundOrder;
-            private Aggregator aggregator;
-
-            /**
-             * Create a new {@code Comparator} for sorting buckets by multiple criteria.
-             *
-             * @param compoundOrder a list of {@link BucketOrder}s to sort on, in order of priority.
-             * @param aggregator    {@link BucketOrder#comparator(Aggregator)}
-             */
-            CompoundOrderComparator(List<BucketOrder> compoundOrder, Aggregator aggregator) {
-                this.compoundOrder = compoundOrder;
-                this.aggregator = aggregator;
-            }
+    private static class SimpleOrder extends InternalOrder {
```

Comment:
Could we add small javadoc on this?  Just thinking that since it's private, someone might some day decide to escalate it public and a description of what it does will either be useful, or convince them not to escalate to public :)

Reply:
:+1: